pkg/test/integration/common/helpers/handling_files.go

```diff
@@ -0,0 +1,230 @@
+				if err != nil {
+					return err
+				}
+			case "Machine":
+				resource := obj.(*v1alpha1.Machine)
+				_, err := c.McmClient.MachineV1alpha1().Machines(namespace).Create(resource)
+				if err != nil {
+					return err
+				}
+			case "MachineDeployment":
+				resource := obj.(*v1alpha1.MachineDeployment)
+				_, err := c.McmClient.MachineV1alpha1().MachineDeployments(namespace).Create(resource)
+				if err != nil {
+					return err
+				}
+			case "Deployment":
+				resource := obj.(*v1.Deployment)
+				_, err := c.Clientset.AppsV1().Deployments(namespace).Create(resource)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	} else {
+		return err
+	}
+	return nil
+}
+
+// checkEstablished uses the specified name to check if it is established
+func (c *Cluster) checkEstablished(crdName string) error {
+	/* This function checks the CRD for an established status
+	if established status is not met, it will sleep until it meets
+	*/
+	err := wait.Poll(500*time.Millisecond, 60*time.Second, func() (bool, error) {
+		crd, err := c.apiextensionsClient.ApiextensionsV1beta1().CustomResourceDefinitions().Get(crdName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		for _, cond := range crd.Status.Conditions {
+			switch cond.Type {
+			case apiextensionsv1beta1.Established:
+				if cond.Status == apiextensionsv1beta1.ConditionTrue {
+					// log.Printf("crd %s is established/ready\n", crdName)
+					return true, err
+				}
+			case apiextensionsv1beta1.NamesAccepted:
+				if cond.Status == apiextensionsv1beta1.ConditionFalse {
+					log.Printf("Name conflict: %v\n", cond.Reason)
+					log.Printf("Naming Conflict with created CRD %s\n", crdName)
+				}
+			}
+		}
+		return false, err
+	})
+	return err
+}
+
+func (c *Cluster) ApplyFiles(source string, namespace string) error {
+	var files []string
+	err := filepath.Walk(source, func(path string, info os.FileInfo, err error) error {
+		files = append(files, path)
+		return nil
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	for _, file := range files {
+		// log.Println(file)
+		fi, err := os.Stat(file)
+		if err != nil {
+			log.Println("Error: file does not exist!")
+			return err
+		}
+
+		switch mode := fi.Mode(); {
+		case mode.IsDir():
+			// do directory stuff
+			log.Printf("%s is a directory.\n", file)
+		case mode.IsRegular():
+			// do file stuff
+			err := c.ApplyFile(file, namespace)
+			if err != nil {
+				//Ignore error if it says the crd already exists
+				if !strings.Contains(err.Error(), "already exists") {
+					log.Printf("Failed to apply yaml file %s", file)
+					//return err
+				}
+			}
+			log.Printf("file %s has been successfully applied to cluster", file)
+		}
+	}
+	return nil
+}
+
+/*
+RotateLogFile takes file name as input and returns a file object obtained by os.Create
+If the file exists already then it renames it so that a new file can be created
+*/
+func RotateLogFile(fileName string) (*os.File, error) {
```

Comment:
I see a lot of Exported functions. However, if you feel that the functions aren't really used outside this package (which is most cases), then please do not export them. Start the naming with small cases in this case.

Reply:
I have made the required changes.